### PR TITLE
Experiment with an empty/mock protocol/protocol handler/audience - not to be merged

### PR DESCRIPTION
**Pull request by andre4i**
The goal here is to have a prototype enabling one of the examples using the web loader with empty quorums.

### CODE REVIEW

File: packages/loader/container-loader/src/container.ts [hunk 1]
```diff
@@ -1561,6 +1584,10 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                     this.deltaManager
                     : undefined;
 
+            if (this.protocolDetails?.connectionHandler !== undefined) {
+                this.protocolDetails?.connectionHandler(this._protocolHandler, details, opsBehind);
```
Comment (vladsud) [inline packages/loader/container-loader/src/container.ts:1588]: Why it's not a method on this._protocolHandler itself?

File: packages/tools/webpack-fluid-loader/src/protocol.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        super();
+
+        for (const pair of quorumSnapshot.values) {
+            const proposal = pair[1];
+            this.proposals.set(proposal.key, proposal.value);
+        }
+    }
+
+    private broadcastClients() {
+        for (const [member, client] of this.members) {
```
Comment (vladsud) [inline packages/tools/webpack-fluid-loader/src/protocol.ts:61]: why does it broadcast all existing clients?

File: packages/tools/webpack-fluid-loader/src/protocol.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    processMessage(message: ISequencedDocumentMessage, _local: boolean): IProcessMessageResult {
+        const emptyResult = {};
+        const clientId = message.clientId;
+        if (this.quorum.getMember(clientId) !== undefined) {
+            return emptyResult;
+        }
+
+        this.quorum.connectLocalClient(clientId, message.sequenceNumber);
```
Comment (vladsud) [inline packages/tools/webpack-fluid-loader/src/protocol.ts:157]: I think the better name for connectLocalClient would be ensureClientExists(), or something like that

File: packages/tools/webpack-fluid-loader/src/protocol.ts [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    ) { }
+
+    public snapshot(): IQuorumSnapshot {
+        return this.initialSnapshot;
+    }
+
+    processMessage(message: ISequencedDocumentMessage, _local: boolean): IProcessMessageResult {
+        const emptyResult = {};
+        const clientId = message.clientId;
+        if (this.quorum.getMember(clientId) !== undefined) {
```
Comment (vladsud) [inline packages/tools/webpack-fluid-loader/src/protocol.ts:153]: I'd re-erange code:
```ts
if (this.quorum.getMember(clientId) == undefined) {
    this.quorum.connectLocalClient(clientId, message.sequenceNumber);
}
return {};
```

File: packages/tools/webpack-fluid-loader/src/protocol.ts (hunk 2 above)
Comment (andre4i) [reply packages/tools/webpack-fluid-loader/src/protocol.ts:61]: this is an experiment left-over

File: packages/tools/webpack-fluid-loader/src/protocol.ts (hunk 3 above)
Comment (andre4i) [reply packages/tools/webpack-fluid-loader/src/protocol.ts:157]: sure, but this is not a class we're going to be actually shipping. This is kind of an example/model.

File: packages/loader/container-loader/src/container.ts (hunk 1 above)
Comment (vladsud) [reply packages/loader/container-loader/src/container.ts:1588]: Ideally we use existing setConnectionState, but change it's signature (and expectations) to allow tri-states in.

**Comment by andre4i** [comment]
Closing this for now as it is not needed.

**Pull request closed without merging**